docs/development-workflow.md

```diff
@@ -51,6 +51,18 @@ or limit it as before with:
 DEBUG=calypso:* make run
 ```
 
+## Clearing browser data
+
+Seeing some strange errors that only happen on _your_ browser? You may need to clear your browser's local storage and database. Calypso saves a lot here and if you've changed the way your data is kept (especially in redux state), it can break things in unepxected ways during the transition.
```

Comment:
I'm adding this from memory of running into issues with it in my recent history. Unfortunately, I can't remember specific cases anymore as it's been a few weeks. 😕 

I think I do remember seeing some cases where a schema wasn't provided for a reducer, but the state was still being persisted anyway.